ast/misc.go

```diff
@@ -365,12 +371,21 @@ func (n *ExecuteStmt) Accept(v Visitor) (Node, bool) {
 // See https://dev.mysql.com/doc/refman/5.7/en/commit.html
 type BeginStmt struct {
 	stmtNode
-	Pessimistic bool
+	Mode string
 }
 
 // Restore implements Node interface.
 func (n *BeginStmt) Restore(ctx *RestoreCtx) error {
-	ctx.WriteKeyWord("START TRANSACTION")
+	switch n.Mode {
+	case Pessimistic:
+		ctx.WriteKeyWord("BEGIN ")
+		ctx.WriteKeyWord(Pessimistic)
+	case Optimistic:
+		ctx.WriteKeyWord("BEGIN ")
+		ctx.WriteKeyWord(Optimistic)
+	default:
+		ctx.WriteKeyWord("START TRANSACTION")
+	}
```

Comment:
Maybe combine the optimistic & pessimistic cases

```go 
if n.Mode == "" {
    ctx.WriteKeyWord("START TRANSACTION")
} else {
    ctx.WriteKeyWord("BEGIN ")
    ctx.WriteKeyWord(n.Mode)
}
```

But now we have 3 transaction options now: (default), pessimistic, optimistic. Why do we need to distinguish the default case from the optimistic mode?

Reply:
When we specify pessimistic mode as the default mode in the config file, we should be able to run an optimistic transaction.